finetuner-workflow/finetuner/finetuner.py

```diff
@@ -648,7 +675,7 @@ def __init__(self, path: str, context_length: int = 2048):
         length_mb = os.stat(path).st_size / (1 << 20)
         num_tokens = self.length * context_length
         self._padding_is_ambiguous = tokenizer.pad_token_id == tokenizer.eos_token_id
-        self._pad_token_id = tokenizer.pad_token_id
+        self._pad_token_id = 0xffff  # uint16 max
```

Comment:
this will be uint16 max if the vocab size is below uint16, but will be uint32 max if it is above that limit. if it is exactly uint16, it will be uint32 as well as we would lack the space to inject a pad token in.